src/relations/postgresql_provider.py

```diff
@@ -78,9 +78,6 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             event.defer()
             return
 
-        if not self.charm.unit.is_leader():
-            return
-
```

Comment:
Dead code, database_requested should only be emitted on the leader: https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/data_interfaces.py#L2878-L2879